onnx/defs/generator/defs.cc

```diff
@@ -523,4 +527,185 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, {batch_size, sample_size});
         }));
 
+static const char* Range_ver11_doc = R"DOC(
+Generate a tensor containing a sequence of numbers that begin at `start` and extends by increments of `delta` 
+up to `limit` (exclusive).)DOC";
+
+template <typename T>
+inline int compute_output_dim_for_range(
+    const TensorProto* start,
+    const TensorProto* limit,
+    const TensorProto* delta) {
+  const auto& start_data = ParseData<T>(start);
+  const auto& limit_data = ParseData<T>(limit);
+  const auto& delta_data =
+      delta != nullptr ? ParseData<T>(delta) : std::vector<T>(1, 1);
+
+  if (start_data.size() != 1 || limit_data.size() != 1 ||
```

Comment:
Scalar in ONNX has empty shape. This basically means that we support both of scalar and tensor with a single element (shape can be `[1]`, `[1, 1, 1, 1, 1]`, and so on. Is this semantic expected?

Reply:
Not sure what you mean. 

In this `if` condition, I only check to make sure the initializers have a single elements (i.e.) scalar - I have in fact not checked the shape and I believe this loose check is fine.

Reply:
I'd prefer strongly-typed setting. If one variable is a scalar, we should use the scalar definition in ONNX. Could you please throw if the shape is not an empty list?

Reply:
Ok, done.